frontend/src/AssignmentView/UploadSubmissionModal.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { FormEvent, useState, useContext } from "react";
+import { useNavigate } from "react-router-dom";
+import Container from "react-bootstrap/Container";
+import Form from "react-bootstrap/Form";
+import Button from "react-bootstrap/Button";
+import Alert from "react-bootstrap/Alert";
+import Stack from "react-bootstrap/Stack";
+import Modal from "react-bootstrap/Modal";
+import { useCookies } from "react-cookie";
+
+import { userContext } from "../Context/UserContext";
+import { StudentAssignmentContext } from "../Context/StudentAssignmentContext";
+
+// This view is a form for professors to add a assignment
+function UploadSubmissionModal() {
+  const studentAssignment = useContext(StudentAssignmentContext);
+  const user = useContext(userContext);
+  const [cookies, setCookies] = useCookies(["jwt"]);
+
+  const [error, setError] = useState("");
+  const nav = useNavigate();
+  const classId = 1;
+  const assignmentId = 1;
+
+  const [show, setShow] = useState(false);
+  const handleClose = () => setShow(false);
+  const handleShow = () => setShow(true);
+
+  function handleSubmission(data: FormData) {
+    return fetch(
+      `http://localhost:8080/class/${classId}/${assignmentId}/upload`,
+      {
+        method: "POST",
+        mode: "cors",
+        headers: {
+          Authorization: cookies.jwt,
+        },
+        body: data,
+      }
+    )
+      .then((response) => {
+        if (response.status == 401) throw "Unauthorized";
+        return response.json();
+      })
+      .catch((e) => {
+        setError(
+          `Failed uploading! Server responded with: ${String(e).replace(
+            "TypeError: ",
+            ""
+          )}`
+        );
+      });
+  }
+
+  function submit(e: FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+    const data = new FormData(e.currentTarget);
+    handleSubmission(data);
+    // TODO: The API call to submit a file is still unimplemented. Will fix when implemented
+    setError((err) => (err ? "" : `File type is not supported`));
+    //nav("/professor/class/");
```

Comment:
Again, the call is implemented. If you mean to describe the behavior being unimplemented, please note it here.

Reply:
haha oops i was rlly tired writing this and missed a lot of small things, my b